demisto_sdk/commands/download/downloader.py

```diff
@@ -712,6 +723,30 @@ def build_pack_content_object(
 
         return content_object
 
+    def get_playbook_id_by_playbook_name(self, playbook_name: str) -> Optional[str]:
+        """
+        extract the playbook id by name,
+        calling the api returns an object that cannot be parsed properly,
+        and its use is only for extracting the id.
+        
+        Args:
+            playbook_name (str): The name of a playbook
+
+        Returns:
+            Optional[str]: The ID of a playbook
+        """
+        endpoint = "/playbook/search"
+        playbooks = demisto_client.generic_request_func(
+            self.client,
+            endpoint,
+            "POST",
+            response_type="object",
+            body={"query": f"name:{playbook_name}"},
+        )[0]["playbooks"]
```

Comment:
let's make it safer.

```suggestion
        response = demisto_client.generic_request_func(
            self.client,
            endpoint,
            "POST",
            response_type="object",
            body={"query": f"name:{playbook_name}"},
        )
        if not response:
            return None
        if not (playbooks := response[0].get("playbooks"):
            return None
```